tests/neg/extension-methods.scala

```diff
@@ -15,4 +15,23 @@ object Test {
     def f2[T]: T = ???        // error: T is already defined as type T
     def f3(xs: List[T]) = ??? // error: xs is already defined as value xs
   }
-}
\ No newline at end of file
+
+  trait Chain{
+      type Tail <: Chain
+  }
+
+  // The initial clauses of right-associative extension methods are not allowed to depend on each other
+  extension [C2 <: c1.Tail](c2: C2) // Not found: c1
+    def !:[C1 <: Chain](c1: C1): c2.Tail = ???
+
+  extension [C1 <: Chain](c1: C1)
+    def ?:[C2 <: c1.Tail](c2: C2): c2.Tail = ??? // Not found: c1
+
```

Comment:
I'm also wondering if we ever discussed allowing type parameter clauses after the extension term parameter clause:
```scala
extension (c1: Chain)[C2 <: c1.Tail]
  def ...
```

Reply:
I believe we deemed it outside the scope of the proposal, in particular here is what's written in the [sip](https://github.com/scala/improvement-proposals/blob/main/content/clause-interleaving.md#lhs-of-extension-methods)

> The left hand side of extension methods remains unchanged, since they only have one explicit term clause, and since the type parameters are very rarely passed explicitly, it is not as necessary to have multiple type clauses there.

It also complicates even more speaking about the shuffling logic for right-associative extension methods (an out-of-date explanation is [here](https://docs.scala-lang.org/scala3/reference/contextual/right-associative-extension-methods.html))